2024-06-17/main.py

```diff
@@ -0,0 +1,201 @@
+import unittest
+import sqlite3
+import uuid
+from pprint import pprint
+from fastapi import FastAPI, APIRouter, HTTPException
+
+# Create our in-memory DB and cursor
+db = sqlite3.connect("file::memory:?cache=shared", check_same_thread=False)
+cur = db.cursor()
+
+# Create your table SQL statement
+def createDB():
+    cur.execute("CREATE TABLE users(id, name, address, email, phone)")
+
+
+# Create your table!
+createDB()
+
+# Instantiate our API/app
+app = FastAPI()
+router = APIRouter(prefix="/api")
+
+# Endpoint for adding a NEW user
+@router.post("/add/user")
+def add_user(name, address=None, email=None, phone=None) -> dict:
+    id = insertUser(name, address, email, phone)
+    return getUser(id)
```

Comment:
Broadly, it would be nice for the utilities to be defined prior to and adjacent to their respective endpoints, for legibility. For example,

```python
def insertUser(name, ...) -> str:
    ...

@router.post("/add/user")
def add_user(name, ...) -> dict:
    ...
```

Reply:
Would be much more legible, will refactor this.

Reply:
Technically I ended up putting endpoint followed by respective method, personal preference 🤷‍♂️